src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -1621,8 +1621,96 @@ static struct attribute_group xmc_attr_group = {
 	.bin_attrs = xmc_bin_attrs,
 };
 
+static ssize_t cmc_image_read(struct file *filp, struct kobject *kobj,
+	struct bin_attribute *attr, char *buf, loff_t off, size_t count)
+{
+	struct xocl_xmc *xmc =
+		dev_get_drvdata(container_of(kobj, struct device, kobj));
+	ssize_t ret = 0;
+
+	if (xmc->mgmt_binary)
+		goto bail;
+
+	if (off >= xmc->mgmt_binary_length)
+		goto bail;
+
+	if (off + count > xmc->mgmt_binary_length)
+		count = xmc->mgmt_binary_length - off;
+
+	memcpy(buf, xmc->mgmt_binary + off, count);
+
+	ret = count;
+bail:
+	return ret;
+}
+
+static size_t image_write(char **image, size_t sz,
+		char *buffer, loff_t off, size_t count)
+{
+	char *tmp_buf;
+	size_t total;
+
+	if (off == 0) {
+		if (*image)
+			vfree(*image);
+		*image = vmalloc(count);
+		if (!*image)
+			return 0;
+
+		memcpy(*image, buffer, count);
+		return count;
+	}
+
+	total = off + count;
+	if (total > sz) {
+		tmp_buf = vmalloc(total);
+		if (!tmp_buf) {
+			vfree(*image);
+			*image = NULL;
+			return 0;
+		}
+		memcpy(tmp_buf, *image, sz);
+		vfree(*image);
+		sz = total;
+	} else
+		tmp_buf = *image;
```

Comment:
please wrap this line w/ "{}" to pass check_patch.

Reply:
done